interview-questions.md

```diff
@@ -8,44 +8,45 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What does calling super do in a Ruby class?
 
-Your answer:
+Your answer: Within the body of a method, it calls to super acts just like a call to that original method. The search for a method body starts in the superclass of the object that was found to contain the original method.
 
 Researched answer:
+You don't have to type super(from, to) – Ruby will assume that by default. But in the definition ( def initialize ) of the constructor you have to define both parameters, since Ruby doesn't – and shouldn't!
 
 2. What is a gem?
 
-Your answer:
+Your answer: GEM (for Graphics Environment Manager) is an operating environment released by Digital Research (DRI) in 1985 for use with the DOS operating system on Intel 8088 and Motorola 68000 microprocessors.
 
-Researched answer:
+Researched answer: They are simply open source libraries that contain Ruby code and are packaged with a little extra data . Using a gem allows a programmer to use the code within the gem in their own program, without explicitly inserting that code.
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: Relational databases are also called Relational Database Management Systems (RDBMS) or SQL databases. Historically, the most popular of these have been Microsoft SQL Server, Oracle Database, MySQL, and IBM DB2.
 
-Researched answer:
+Researched answer: All relational databases can be used to manage transaction-oriented applications (OLTP), and most non-relational databases, in the categories of Document Stores and Column Stores, can also be used for OLTP, adding to the confusion between them. OLTP databases can be thought of as “operational” databases, characterized by frequent, short transactions that include updates, touch a small amount of data, and provide concurrency to thousands (if not more) of transactions (some examples include banking applications and online reservations).
 
 4. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: A primary key is the column or columns that contain values that uniquely identify each row in a table.A database table must have a primary key for Optim to insert, update, restore, or delete data from a database table
 
-Researched answer:
+Researched answer: The main purpose of a primary key is to implement a relationship between two tables in a relational database; it's not called a relational database for nothing! More specifically, the primary key is the "target" which a foreign key can reference.
```

Comment:
I love the mention of foreign keys here! It adds to the initial answer where you correctly stated that a primary key uniquely identifies each row in a table. 👏🏽